### ISSUE TITLE
Feature request. Small but useful. Entity class.

### ISSUE THREAD

**Issue by nowackipawel** [bcit-ci/CodeIgniter4#1176]
There is an feature which allows to cast value of attribute of entity.
Could someone add another options to switch in Entity::castAs($value, string $type)
 i.e. like this:
```
			case 'json-array':
				if (is_string($value) && (strpos($value, '[') === 0 || strpos($value, '{') === 0))
				{
					$value = json_decode($value, true);
				}
				break;
			case 'json-object':
				if (is_string($value) && (strpos($value, '[') === 0 || strpos($value, '{') === 0))
				{
					$value = json_decode($value, false);
				}
				break;
  ```
imho it will be useful.

**Comment by InsiteFX** [comment]
You should ask this in the CodeIgniter CI 4 Forums not here, this is only for reporting bugs etc;

**Comment by lonnieezell** [comment]
I'm not opposed to this. Though I believe there are some security risks when storing json values as an array and not an object, so we might consider not checking for square brackets, just curly braces. 

Care to submit a PR?

**Comment by jim-parry** [comment]
Was this not addressed already, in #1283 ?

**Comment by lonnieezell** [comment]
Yes, it did.
